### DISCUSSION TITLE
navigation.openDrawer() works but navigation.closeDrawer() does not

### DISCUSSION THREAD

**Discussion by welschmoor** [expo/expo#27564] [Archived: Questions]
Expo Router V3
```
import { useNavigation } from 'expo-router'

//this command works:
 navigation?.openDrawer()
 
//this does not:
 navigation?.closeDrawer()
```
nvigation.dispatch(DrawerActions... does not work either.

A temporary solution is using the { navigation } prop like we used to in react-navigation that the drawer receives, instead of navigation = useNavigation(). Another one that works is router.back(), since drawer is a route. 

How do I programmatically close the drawer?

**Comment by 121Unicorns** [comment]
I have the same problem. I was previously using the ````DrawerToggleButton```` from ````@react-navigation/drawer```` but this now throws the error that toggleDrawer was not handled by any navigator.

**Reply by Alex44lel** [reply]
Did you manage to solve this?

I am not able to customize the drawer hamburguer button as there is no way to open or close the meny.

I have tried:

 
`navigation.dispatch(DrawerActions.toggleDrawer())
`
and

`navigation.toggleDrawer()
`
none of them work

**Reply by PiotrekSy** [reply]
The workaround i have found was passing 'pathname' variable as Your dynamic drawer. You already have it in drawer component.
  🚀   🚀   🚀   🚀   🚀   🚀   🚀   🚀   🚀   🚀 
  
 !!!Works if You use file based routing.!!!
  
```
 
// !! DOES NOT WORK 
const navigation = useNavigation();
<Pressable onPress={() => navigation.goBack()}>


//  Works fine: 
const pathname = usePathname();
<Pressable onPress={() => router.push(pathname)}>

Hides drawer and 
 
```

**Reply by 121Unicorns** [reply]
In the layout page where the header would be called, I used:

```
import { useNavigationContainerRef} from "expo-router";
const navigation = useNavigationContainerRef();
```

Then I passed navigation into the custom component such that:

```<CustomHeader navigation={navigation} />```

My custom component is defined as: 

```export default function CustomHeader({ navigation }: any) {...}```

Then in the button I wanted to handle drawer opening/closing, I used:

```
<TouchableOpacity
                onPress={() => {
                  navigation.dispatch({ type: "TOGGLE_DRAWER" });
                }}
              >
                <LeftArrowIcon size={24} color={"#FEFEFE"} />
              </TouchableOpacity>
```

Everything worked great after that.

**Reply by daveclinton** [reply]
This works

**Reply by stanleyugwu** [reply]
This works, and I didn't pass the navigation as prop, I called the hook directly inside the component I'm closing the drawer from. I had to cause of memoization and it still worked.

```tsx
...
const CustomDrawerContent = memo(() => {
  const router = useRouter();
  const navigation = useNavigationContainerRef();

  const navigateToSection = (section:Section) => {
    navigation.dispatch(DrawerActions.closeDrawer());
    router.push(section);
  };
})
```

**Comment by VimukthiShohan** [comment]
+1

**Comment by RoyRao2333** [comment]
You can try

```tsx
import { DrawerActions } from '@react-navigation/native';
import { useNavigation } from 'expo-router';

navigation.dispatch(DrawerActions.closeDrawer());
```

Works for me 😊
